Add setup script and template files

ewquon opened the pull request:
Files courtesy of @phawbeck.

Changes:
- Remove `$HOME_DIR` parameter, instead storing template files in a subdirectory
- Remove hard-coded email/passwords
- Add prompt for email/pass

ewquon committed:
Generalize and clean up (minor) setup script

ewquon committed:
Make ICBC_DIR if it doesn't exist

Use -p flag for mkdir to avoid error if parent directories don't exist

ewquon committed:
Add option `DOM_STR` to select WPS input template

ewquon committed:
Move templates from HOME_DIR into tracked templates subdir

- Update setup.sh to use tracked templates
- Switch reanalysis dataset to GFS

ewquon committed:
Use full path for TEMPLATE_DIR

ewquon committed:
Prompt for RDA email/pass

After running the reanalysis data retrieval script, the script
(containing the password in plain text) is removed.

ewquon committed:
Load computing environment at beginning of script (in user section)

ewquon committed:
Restore default environment for testing on Eagle

ewquon committed:
Merge branch 'master' into feature/cleanup_setup_templates

hawbecker commented on setup.sh:
```diff
@@ -27,37 +28,55 @@ ddS="08"
 # NUMBER OF DOMAINS TO USE
 MAX_DOM="2"
+# DOMAIN CONFIG (default: "")
+DOM_STR=""
```
When would this be needed?

ewquon replied:
To select `namelist.wps.template*`

hawbecker commented on setup.sh:
```diff
@@ -80,23 +106,23 @@ do
-        ./get_erai.csh
-        cp get_erai.csh get_erai.csh_$CASE_STR"_"$counter
+        sed "s/SETPASSWD/$passwd/g" $TEMPLATE_DIR/get_erai_template.csh > get_erai.csh
+        sed -i "s/MY.EMAIL@asdf.com/$emailaddr/g" get_erai.csh
```
I don't know how when I was searching for "how to use sed" this never came up. Everything I saw was saying you should save to a new file name every time... this is much better.

hawbecker commented on setup.sh:
```diff
@@ -124,9 +150,16 @@ echo "Finished downloading ICBC data."
+WPS_TEMPLATE="$TEMPLATE_DIR/namelist.wps.template"
+if [ -n "$DOM_STR" ]; then
+    # append DOM_STR
+    WPS_TEMPLATE="${WPS_TEMPLATE}.${DOM_STR}"
```
Typically you'll only be running WPS once for all domains.

hawbecker reviewed (commented):
Left a few comments on the DOM_STR - just not sure what this would be used for.

hawbecker reviewed (approved):
I understand the "DOM_STR" variable now to be what selects something like "namelist.wps.template.large" or any other variable namelist template.

twjuliano reviewed (approved):
I think we need the following command somewhere for the WRF run:

`cp myoutfields.txt $OUT_DIR
`

Otherwise, namelist.input points to a txt file that is not in the newly-created folder.

ewquon committed:
Get rid of OUT_DIR parameter

The repository is supposed to track namelist.input, so it doesn't make sense
if the generated namelist (and other files) end up outside of the repo

This also addresses the issue brought up by @twjuliano

ewquon committed:
Track namelist.wps instead of namelist.wps.template

ewquon committed:
Create .gitignore to exclude outputs from setup.sh

ewquon committed:
Further script cleanup

ewquon commented:
I've fixed the issue brought up by @twjuliano. Also, I changed the workflow a bit, to something more in line with what I had originally imagined. 

My original intention was to track namelist.input (and any auxiliary WRF input files); the namelist.wps and other files should be straightforward to generate manually for an experienced WRF user. However, after adding the nice `setup.sh` script that @phawbeck provided, I realized that all generated input files would be placed outside of the repository and therefore not be tracked (per my original intention). The new workflow follows what I had originally outlined in the README.

ewquon committed:
Properly generate submit_wrf.sh

ewquon committed:
Filter out wrf output files from repo
